playwright/pages/components/EnvironmentSelector.ts

```diff
@@ -0,0 +1,61 @@
+import { expect, Locator, Page } from "@playwright/test";
+
+export class EnvironmentSelector {
+  readonly page: Page;
+  readonly loginTextButton: Locator;
+  readonly loginIconButton: Locator;
+  readonly environmentName: Locator;
+  readonly dropdownTrigger: Locator;
+
+  constructor(page: Page) {
+    /**
+     * Components
+     */
+    this.page = page;
+
+    /**
+     * Static locators
+     */
+    this.loginTextButton = page.getByRole("button", {
+      name: "Login required",
+      exact: true,
+    });
+    // TODO: Replace with `getByRole` once `<IconButton>` supports labels.
+    this.loginIconButton = page.getByTestId("environment-login-icon-button");
+    this.environmentName = page.getByTestId("active-environment-name");
+    // TODO: Replace with `getByRole` once `<IconButton>` supports labels.
+    this.dropdownTrigger = page.getByTestId("environment-dropdown-button");
+  }
+
+  /**
+   * Dynamic locators
+   */
+  getEnvironmentMenuItem(name: string): Locator {
+    return this.page.getByRole("menuitem", { name, exact: true });
+  }
+
+  /**
+   * Actions
+   */
+  async selectEnvironment(name: string) {
+    await this.dropdownTrigger.click();
+    await this.getEnvironmentMenuItem(name).click();
+    await expect(this.environmentName).toHaveText(name);
+  }
+
+  /**
+   * Assertions
+   */
+  // TODO: Type `kind` with `@slicemachine/manager`'s `Environment["kind"]`.
+  async checkSelectedEnvironmentKind(kind: "prod" | "stage" | "dev") {
```

Comment:
❓ I was a big confused with the naming, maybe we could be specific to what you're testing, the dot, no?

Reply:
I originally tried to test more for behavior rather than the specific dot, but it didn't really work well. I changed the implementation to `checkHasSelectedEnvironmentDotKind()` with the following:

```ts
  async checkHasSelectedEnvironmentDotKind(kind: "prod" | "stage" | "dev") {
    const humanReadableKindMap = {
      prod: "Production",
      stage: "Staging",
      dev: "Development",
    } as const;

    await expect(this.page.getByTestId("active-environment-dot")).toBeVisible();
    await expect(this.page.getByTestId("active-environment-dot")).toHaveText(
      `${humanReadableKindMap[kind]} environment`,
    );
  }
```

Reply:
(I'm resolving the conversation so we can track what was addressed, but feel free to re-open it if you have any other suggestions!)